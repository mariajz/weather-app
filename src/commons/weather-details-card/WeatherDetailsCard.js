import React from 'react';
import { View } from 'react-native';
import { StyledText } from '../styles';
import { CustomIcon } from '../visual-elements';
import {
    CardWrapper,
    ContentWrapper,
    HeadingWrapper,
} from './WeatherDetailsCard.style';

const WeatherDetailsCard = ({ weatherDetailsCardProps }) => {
    const {
        iconName,
        cardName,
        title,
        subtitle = '',
        content = '',
        Visual = <></>,
    } = weatherDetailsCardProps;
    return (
        <CardWrapper>
            <HeadingWrapper>
                <CustomIcon size={15} iconName={iconName} />
                <StyledText marginLeft={5} size={14}>
                    {cardName}
                </StyledText>
            </HeadingWrapper>
            <ContentWrapper>
                <View>
                    <StyledText size={28}>{title}</StyledText>
                    <StyledText size={20}>{subtitle}</StyledText>
                </View>

                <View>{Visual}</View>
                <StyledText size={10}>{content}</StyledText>
            </ContentWrapper>
        </CardWrapper>
    );
};

export default WeatherDetailsCard;
