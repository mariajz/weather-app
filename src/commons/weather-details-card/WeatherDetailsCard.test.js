import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import WeatherDetailsCard from './WeatherDetailsCard';

const weatherDetailsCardProps = {
    iconName: 'humidity',
    cardName: 'HUMIDITY',
    title: '51%',
    subtitle: 'Moderate',
    content: 'Pleasent weather',
    Visual: <></>,
};

jest.mock('../../commons/visual-elements', () => ({
    CustomIcon: () => <MockView />,
    Theme: {
        BackgroundWhite: jest.fn(
            opacity => `mocked rgba(255, 255, 255, ${opacity})`,
        ),
    },
}));

describe('WeatherDetailsCard', () => {
    it('should render WeatherDetailsCard', () => {
        const container = render(
            <WeatherDetailsCard
                weatherDetailsCardProps={weatherDetailsCardProps}
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should render WeatherDetailsCard when subtitle and content are optional', () => {
        const modifiedProps = {
            iconName: 'humidity',
            cardName: 'HUMIDITY',
            title: '51%',
            Visual: <></>,
        };
        const container = render(
            <WeatherDetailsCard weatherDetailsCardProps={modifiedProps} />,
        );

        expect(container).toMatchSnapshot();
    });
});
