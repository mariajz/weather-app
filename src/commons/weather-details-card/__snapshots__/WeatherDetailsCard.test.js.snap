// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetailsCard should render WeatherDetailsCard 1`] = `
<View
  style={
    {
      "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "49%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      HUMIDITY
    </Text>
  </View>
  <View
    style={
      {
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 8,
      }
    }
  >
    <View>
      <Text
        size={28}
        style={
          {
            "color": "white",
            "fontSize": 28,
            "marginLeft": 0,
          }
        }
      >
        51%
      </Text>
      <Text
        size={20}
        style={
          {
            "color": "white",
            "fontSize": 20,
            "marginLeft": 0,
          }
        }
      >
        Moderate
      </Text>
    </View>
    <View />
    <Text
      size={10}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginLeft": 0,
        }
      }
    >
      Pleasent weather
    </Text>
  </View>
</View>
`;

exports[`WeatherDetailsCard should render WeatherDetailsCard when subtitle and content are optional 1`] = `
<View
  style={
    {
      "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "49%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      HUMIDITY
    </Text>
  </View>
  <View
    style={
      {
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 8,
      }
    }
  >
    <View>
      <Text
        size={28}
        style={
          {
            "color": "white",
            "fontSize": 28,
            "marginLeft": 0,
          }
        }
      >
        51%
      </Text>
      <Text
        size={20}
        style={
          {
            "color": "white",
            "fontSize": 20,
            "marginLeft": 0,
          }
        }
      />
    </View>
    <View />
    <Text
      size={10}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginLeft": 0,
        }
      }
    />
  </View>
</View>
`;
